flowmachine/tests/test_label_event_score.py

```diff
@@ -0,0 +1,233 @@
+            ],
+        },
+        "day": {
+            "type": "Polygon",
+            "coordinates": [[[-1, -0.5], [-1, 0.5], [0, 0.5], [0, -0.5]]],
+        },
+    }
+    assert not LabelEventScore.bounds_dict_has_overlaps(
+        LabelEventScore._make_bounds_dict(bounds_dict)
+    )
+
+
+def test_overlaps_bounds_dict_raises():
+    """
+    Should raise an error if an overlap is found.
+    """
+    bounds_dict = {
+        "evening": {
+            "type": "MultiPolygon",
+            "coordinates": [
+                [[[0.000_001, 0.5], [0.000_001, 1], [1, 1], [1, 0.5]]],
+                [[[0.000_001, -1], [0.000_001, -0.5], [1, -0.5], [1, -1]]],
+            ],
+        },
+        "day": {
+            "type": "Polygon",
+            "coordinates": [[[-1, -0.5], [-1, 0.75], [0, 0.5], [0.1, -0.75]]],
+        },
+    }
+    with pytest.raises(ValueError):
+        LabelEventScore.bounds_dict_has_overlaps(
+            LabelEventScore._make_bounds_dict(bounds_dict)
+        )
+
+
+def test_constructor_overlaps_bounds_dict_raises():
+    """
+    Should raise an error if an overlap is found in constructor.
+    """
+    bounds_dict = {
+        "evening": [
+            {
+                "hour_lower_bound": 0,
+                "hour_upper_bound": 1,
+                "day_of_week_upper_bound": 1,
+                "day_of_week_lower_bound": 0.5,
+            },
+            {
+                "hour_lower_bound": 0.00001,
+                "hour_upper_bound": 1,
+                "day_of_week_upper_bound": -0.5,
+                "day_of_week_lower_bound": -1,
+            },
+        ],
+        "day": [
+            {
+                "hour_lower_bound": -1,
+                "hour_upper_bound": 0,
+                "day_of_week_upper_bound": 0.5,
+                "day_of_week_lower_bound": -0.5,
+            }
+        ],
+    }
+    with pytest.raises(ValueError):
+        LabelEventScore(
+            scores=EventScore(start="2016-01-01", stop="2016-01-05"), labels=bounds_dict
+        )
+
+
+# Test bad bounds error handling
+
+bad_bounds = [
+    ({}, ValueError),
+    ("BAD", ValueError),
+    (
+        {
+            "type": "NOT_A_TYPE_OF_SHAPE",
+            "coordinates": [
+                [[[0.000_001, 0.5], [0.000_001, 1], [1, 1], [1, 0.5]]],
+                [[[0.000_001, -1], [0.000_001, -0.5], [1, -0.5], [1, -1]]],
+            ],
+        },
+        ValueError,
+    ),
+    (
+        {
+            "type": "NOT_A_TYPE_OF_SHAPE",
+            "coordinates": [
+                [0.000_001, 0.5],
+                [0.000_001, 1],
+                [1, 1],
+                [1, 0.5],
+                [[[0.000_001, -1], [0.000_001, -0.5], [1, -0.5], [1, -1]]],
+            ],
+        },
+        ValueError,
+    ),
+]
+
+
+@pytest.mark.parametrize("bad_bound", next(zip(*bad_bounds)))
+def test_constructor_raises_value_error(bad_bound):
+    """
+    Constructor should only raise value errors.
```

Comment:
Not opposed to this, but out of curiosity, what's the reason we (a) want this and (b) explicitly need to test for it?